src/main/java/com/couchbase/lite/replicator/PullerInternal.java

```diff
@@ -515,8 +515,6 @@ public void insertDownloads(List<RevisionInternal> downloads) {
                     }
                 }
 
-                if(rev.getBody() != null) rev.getBody().release();
-
```

Comment:
`Body.compact()` is called before reaching this line. So `Body` holds `json` value. I believe iOS implementation also holds `json` value at this point. However, as you mentioned, the peak memory usage might increase by deleting this line.